### ISSUE TITLE
Including CSS style files within SVG

### ISSUE THREAD

**Issue by edemaine** [svgdotjs/svg.js#517]
SVG supports a [style tag](https://www.w3.org/TR/SVG/styling.html#StyleElement) where you can embed CSS to be referenced by tag's classes.  You already support adding classes. Could we get a helper to add `style` tags to the top-level SVG (as in [this example](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/style)) and/or to the defs section (as in [this example](https://www.w3.org/TR/SVG/styling.html#StyleElementExample))?  I'd be fine with the argument just being a CSS string...

**Comment by Fuzzyma** [comment]
I answered a question about that on stackoverflow: http://stackoverflow.com/questions/35681530/specify-font-from-font-file-in-svg-using-svg-js/35681770#35681770

I altered the code a bit:

```javascript

SVG.Style = SVG.invent({
  // Initialize node
  create: 'style'

  // Inherit from
, inherit: SVG.Element

  // Add class methods
, extend: {

  , style: function(style){
      if(style == null) return this.node.textContent
      this.node.textContent = style
      return this
    }

  }

, construct: {
    // Create a style element
    style: function(style) {
      return this.put((new SVG.Style()).style(style))
    }
  }

})
```

So you would go this way:
`draw.defs().style('#myId{fill:red}')`


Of course advanced rule handling would be better - maybe later ^^

**Comment by edemaine** [comment]
Had occasion to do this again.  `svg.element('style').words('...')` works, though it'd still be nice to have a more intuitive helper here.  I guess the tricky part is naming --- `.style` already means apply a style, not access a `<style>` tag.  I guess `.defs().style` is a fine way to do it.

**Comment by Fuzzyma** [comment]
Yes... Naming clash... W it could be named `styleTag` instead. However I am not sure about what you are asking

**Comment by dotnetCarpenter** [comment]
Ref: https://github.com/svgdotjs/svg.js/issues/579

**Comment by wout** [comment]
@Fuzzyma `.stylesheet()` would be better than `.styleTag()` I think.

**Comment by yyynnn** [comment]
Using the `Bare` for styling. I was searching for <style> element in docs, thought it was essential, well, it is essential, isn't it?
Better not to include it in the core, but maybe one could add some emphasis in the docs at [https://svgdotjs.github.io/manipulating/styles/](url), about how `svg.element` can help in making style tag. A little extension if you will.

**Comment by wout** [comment]
@yyynnn We should make this a plugin I think and reference it in the style page. It's an issue of time for us, so all help is welcome ;)

**Comment by Fuzzyma** [comment]
Can be tried in the [style-tags](https://github.com/svgdotjs/svg.js/tree/517-style-tags) branch. Pull, build with `npm build` and try this:

```js

let canvas = SVG().addTo(containerDiv).size(1000, 500)
canvas.style('#selector', {
  opacity: 0.1,
  ...
})

// or

canvas.style('#selector{opacity:0.1}')

// or

canvas.style().rule('#selector', {
  opacity: 0.1,
  ...
})

// or

canvas.font('name', 'src')

// or all together

canvas.style(...).rule(...).rule(...).font(...).rule(...).font(...)

**Comment by Fuzzyma** [comment]
landed in 3.0 with 24111571324badc5282cf835c6565576d7fcb90f
